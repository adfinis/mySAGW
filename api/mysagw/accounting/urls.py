from django.urls import path

from . import views

urlpatterns = [
    path("receipts/<uuid:pk>", views.ReceiptView.as_view(), name="receipts"),
]
