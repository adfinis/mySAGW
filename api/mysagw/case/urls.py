from rest_framework.routers import SimpleRouter

from . import views

r = SimpleRouter(trailing_slash=False)

r.register(r"case/accesses", views.CaseAccessViewSet)
r.register(r"downloads", views.CaseDownloadViewSet, basename="downloads")

urlpatterns = r.urls
