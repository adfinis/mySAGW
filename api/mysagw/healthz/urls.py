from django.urls import re_path

from . import views

urlpatterns = [
    re_path(r"healthz", views.HealthzView.as_view(), name="healthz"),
]
