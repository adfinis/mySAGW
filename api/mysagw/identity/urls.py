from django.urls import re_path
from rest_framework.routers import SimpleRouter

from . import views

r = SimpleRouter(trailing_slash=False)

r.register(r"additional-emails", views.EmailViewSet)
r.register(r"phone-numbers", views.PhoneNumberViewSet)
r.register(r"addresses", views.AddressViewSet)
r.register(r"identities", views.IdentityViewSet)
r.register(
    r"public-identities",
    views.PublicIdentitiesViewSet,
    basename="public-identities",
)
r.register(r"interest-categories", views.InterestCategoryViewSet)
r.register(r"interests", views.InterestViewSet)
r.register(r"membership-roles", views.MembershipRoleViewSet)
r.register(r"memberships", views.MembershipViewSet)
r.register(r"my-orgs", views.MyOrgsViewSet, basename="my-orgs")
r.register(r"my-memberships", views.MyMembershipViewSet, basename="my-memberships")
r.register(
    r"org-memberships",
    views.OrganisationAdminMembersViewSet,
    basename="org-memberships",
)

urlpatterns = [
    re_path(
        r"^me/?$",
        views.MeViewSet.as_view({"get": "retrieve", "patch": "update"}),
        name="me",
    ),
]

urlpatterns.extend(r.urls)
