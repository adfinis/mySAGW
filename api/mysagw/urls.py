from django.conf.urls import include
from django.urls import path

urlpatterns = [
    path("api/v1/", include("mysagw.identity.urls")),
    path("api/v1/", include("mysagw.snippets.urls")),
    path("api/v1/", include("mysagw.accounting.urls")),
    path("api/v1/", include("mysagw.case.urls")),
    path("api/v1/", include("mysagw.healthz.urls")),
]
