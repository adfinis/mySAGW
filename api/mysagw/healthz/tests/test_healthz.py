import pytest
from django.urls import reverse
from rest_framework import status
from watchman import settings as watchman_settings


def _set_identity_healthcheck_properties(identity):
    identity.email = "sagw@sagw.ch"
    identity.is_organisation = True
    identity.save()


@pytest.mark.parametrize(
    "client,expected_status",
    [
        ("user", status.HTTP_403_FORBIDDEN),
        ("staff", status.HTTP_200_OK),
        ("admin", status.HTTP_200_OK),
    ],
    indirect=["client"],
)
def test_healthz_permissions(db, snapshot, identity, client, expected_status):
    _set_identity_healthcheck_properties(identity)

    url = reverse("healthz")

    response = client.get(url)

    assert response.status_code == expected_status

    if expected_status != status.HTTP_200_OK:
        return

    snapshot.assert_match(response.json())


def test_db_migrations_not_applied(
    capsys,
    client,
    snapshot,
    db,
    roll_back_migrations,
    identity,
):
    """
    Test /healthz/ endpoint response when migrations haven't been applied.

    Undo applied migrations for app 'contenttypes'.
    Affects database migrations health check, which should detect unapplied
    migrations and fail.
    """
    _set_identity_healthcheck_properties(identity)

    # get /healthz/ response and compare to previous snapshot
    response = client.get(reverse("healthz"))
    snapshot.assert_match(response.json())

    # assert database migrations check fails
    assert response.json()["database migrations"] == [{"default": {"ok": False}}]
    assert response.status_code == watchman_settings.WATCHMAN_ERROR_CODE
