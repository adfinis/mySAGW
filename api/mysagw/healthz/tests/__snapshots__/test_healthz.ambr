# serializer version: 1
# name: test_db_migrations_not_applied[admin]
  dict({
    'caches': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'database migrations': list([
      dict({
        'default': dict({
          'ok': False,
        }),
      }),
    ]),
    'database models': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'databases': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
  })
# ---
# name: test_healthz_permissions[admin-200]
  dict({
    'caches': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'database migrations': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'database models': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'databases': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
  })
# ---
# name: test_healthz_permissions[staff-200]
  dict({
    'caches': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'database migrations': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'database models': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
    'databases': list([
      dict({
        'default': dict({
          'ok': True,
        }),
      }),
    ]),
  })
# ---
